tests/components/dnsip/test_config_flow.py

```diff
@@ -0,0 +1,247 @@
+            result["flow_id"],
+            {
+                CONF_HOSTNAME: "home-assistant.io",
+            },
+        )
+        await hass.async_block_till_done()
+
+    assert result2["type"] == RESULT_TYPE_CREATE_ENTRY
+    assert result2["title"] == "home-assistant.io"
+    assert result2["data"] == {
+        "hostname": "home-assistant.io",
+        "name": "home-assistant.io",
+        "resolver": "208.67.222.222",
+        "resolver_ipv6": "2620:0:ccc::2",
+        "ipv4": True,
+        "ipv6": True,
+    }
+    assert len(mock_setup_entry.mock_calls) == 1
+
+
+async def test_form_error(hass: HomeAssistant) -> None:
+    """Test validate url fails."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    with patch(
+        "homeassistant.components.dnsip.config_flow.aiodns.DNSResolver",
+        side_effect=DNSError("Did not find"),
+    ):
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {
+                CONF_HOSTNAME: "home-assistant.io",
+            },
+        )
+        await hass.async_block_till_done()
+
+    assert result2["type"] == RESULT_TYPE_FORM
+    assert result2["step_id"] == "user"
+    assert result2["errors"] == {"base": "invalid_hostname"}
+
+
+@pytest.mark.parametrize(
+    "p_input,p_output",
+    [
+        (
+            {CONF_HOSTNAME: "home-assistant.io"},
+            {
+                "hostname": "home-assistant.io",
+                "name": "home-assistant.io",
+                "resolver": "208.67.222.222",
+                "resolver_ipv6": "2620:0:ccc::2",
+                "ipv4": True,
+                "ipv6": True,
+            },
+        ),
+        (
+            {},
+            {
+                "hostname": "myip.opendns.com",
+                "name": "myip",
+                "resolver": "208.67.222.222",
+                "resolver_ipv6": "2620:0:ccc::2",
+                "ipv4": True,
+                "ipv6": True,
+            },
+        ),
+    ],
+)
+async def test_import_flow_success(
+    hass: HomeAssistant, p_input: dict[str, str], p_output: dict[str, str]
+) -> None:
+    """Test a successful import of YAML."""
+
+    with patch(
+        "homeassistant.components.dnsip.config_flow.aiodns.DNSResolver",
+        return_value=RetrieveDNS(),
+    ), patch(
+        "homeassistant.components.dnsip.async_setup_entry",
+        return_value=True,
+    ) as mock_setup_entry:
+        result2 = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": config_entries.SOURCE_IMPORT},
+            data=p_input,
+        )
+        await hass.async_block_till_done()
+
+    assert result2["type"] == RESULT_TYPE_CREATE_ENTRY
+    assert result2["title"] == p_output["name"]
+    assert result2["data"] == p_output
+    assert len(mock_setup_entry.mock_calls) == 1
+
```

Comment:
Please add a test that the flow aborts when an existing config entry is present for the same hostname

Reply:
Good catch. Forgot to add it. Now it's done 👍